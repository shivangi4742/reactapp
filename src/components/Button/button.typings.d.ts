import { ReactNode } from "react";

export type ButtonProps = {
    children: ReactNode,
    onClick?: () => void,
    className?: string, 
    type?: string,
    size?: string,
    disabled?: boolean
}