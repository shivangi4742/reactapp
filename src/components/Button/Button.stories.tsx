import * as React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './button';
import { Grid } from '@material-ui/core';
import ComponentStories from '../component-stories';
import ComponentAPI from '../component-api';
import Storiesheader from '../stories-header';

const stories = storiesOf('Components', module);

const codeString = `
<Button type='primary'> Primary </Button>
<Button>Default</Button>
<Button disable> Disable </Button>
 <Button type="dashed">Dashed</Button>
 <Button type="danger">Danger</Button>
`;
const codeString1 = `
    <Button type='primary' size='sm'>
        Small
    </Button>
    <Button size='md'>
        Medium
    </Button>
    <Button size='lg'>
        Large
    </Button>
`;
const codeString2 = `
 <Button type='share' >
     share
   </Button>
<Button type='notification'>
        1
</Button>
 <Button type='remarks'>
   Add remarks
</Button>`;
const rows = [
    {
        id: 0,
        property: 'size',
        description: 'can be set to small as (sm) large as (lg) or omitted',
        type: 'string',
        default: "-"
    },
    {
        id: 1,
        property: 'type',
        description: 'can be set to primary ghost dashed danger or omitted (meaning default)',
        type: 'string',
        default: "-"
    }

]
stories.add(
    'Button',
    (() => <div>
        <Storiesheader heading="Button" shortdesc="To trigger an operation." desc="A button means an operation (or a series of operations). Clicking a button will trigger corresponding business logic."></Storiesheader>
        <h3>Examples <a href="#examples">#</a></h3>
        <Grid container spacing={40}>
            <ComponentStories syntax={codeString} >
                <Button type='primary'> Primary </Button>{' '}
                <Button>Default</Button>{' '}
                <Button disabled>   Disable </Button>{' '}
                <Button type="dashed">Dashed</Button>{' '}
                <Button type="danger">Danger</Button>
            </ComponentStories>
            <ComponentStories syntax={codeString1} >
                <Button type='primary' size='sm'>
                    Small
               </Button> {' '}
                <Button size='md'>
                    Medium
               </Button> {' '}
                <Button size='lg'>
                    Large
                </Button>
            </ComponentStories>
            <ComponentStories syntax={codeString2} >
                <Button type='share' >
                    share
                  </Button> {' '}
                <Button type='notification'>
                    1
              </Button> {' '}
                <Button type='remarks'>
                    Add remarks
                </Button>
            </ComponentStories>

        </Grid>
        <ComponentAPI rows={rows} />
    </div>
    )
);