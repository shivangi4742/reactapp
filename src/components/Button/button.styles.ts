import Styled, { css } from 'styled-components';
import DS from '../../utils/design-system';

const Button = Styled.button`
    border: none;
    background: transparent;
    text-align: center;
    border-radius: ${DS.BORDER_RADIUS};
    border: 1px solid ${DS.PALETTE.gray}
    color: ${DS.PALETTE.black2};
    width: auto;
    height: ${DS.BUTTON.DEFAULT.height}px;
    line-height: ${DS.BUTTON.DEFAULT.height}px;
    font-size: ${DS.BUTTON.DEFAULT.fontSize}px;
    text-transform: capitalize;
    padding: 0 15px;
    cursor: pointer;
   ${props => props.type === 'primary' && css`
        background: ${DS.PALETTE.BLUE};
        border: 1px solid transparent;
        color: ${DS.PALETTE.WHITE};
   `}
    ${props => props.type === 'dashed' && css`
          color: rgba(0,0,0,0.65);
        background-color: #fff;
        border-color: #d9d9d9;
        border-style: dashed;
          &:hover {
              color : #1890ff;;
            border : dashed #1890ff;;
    }
   `}
    ${props => props.type === 'danger' && css`
      color: #f5222d;
        background-color: #f5f5f5;
        border-color: #d9d9d9;
        &:hover {
             background-color:#f5222d;
             color : #fff;
            box-shadow: rgba(0,0,0,.2) 0 2px 6px 0;
    }
   `}
   ${props => props.size === 'sm' && css`
        width: auto;
        height: ${DS.BUTTON.SMALL.height}px;
        line-height: ${DS.BUTTON.SMALL.height}px;
        font-size: ${DS.BUTTON.SMALL.fontSize}px;
   `}
   ${props => props.size === 'lg' && css`
       width: auto;
        height: ${DS.BUTTON.LARGE.height}px;
        line-height: ${DS.BUTTON.LARGE.height}px;
        font-size: ${DS.BUTTON.LARGE.fontSize}px;
        
   `}
    ${props => props.type === 'share' && css`
        width: 96px;
        height: 40px;
        color: ${DS.PALETTE.WHITE}
        background-color: #1194f6;
       font-family: ${DS.FONT_FAMILY.REGULAR};
        text-transform: uppercase;
        font-size: 14px;
       font-weight: 500;
   `}
    ${props => props.type === 'notification' && css`
        width: 25px;
        height: 24px;
        color: #212121;
       background-color: ${DS.PALETTE.GRAY};
        line-height: 23px;
        font-size: 14px;
      font-family:${DS.FONT_FAMILY.REGULAR};
   `}
    ${props => props.type === 'remarks' && css`
        width: 110px;
        height: 36px;
        color: rgba(0, 0, 0, 0.87);
        background-color: ${DS.PALETTE.GRAY};
        font-family: ${DS.FONT_FAMILY.REGULAR};
        font-size: 12px;
        font-weight: 700;
        line-height: 14px;
        text-align: center; 
   `}
   ${props => props.disable && css`
       border: 1px solid #999999;
    background-color: #cccccc;
    color: #666666;
    cursor: not-allowed;
   `}
`
export default Button;