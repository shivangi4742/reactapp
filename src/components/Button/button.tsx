import React from 'react';
import { ButtonProps } from './button.typings';
import StyledBtn from './button.styles';

const Button = ({ children, ...props }: ButtonProps) => {
    return (
        <StyledBtn {...props}>{children} </StyledBtn>
    )
}

export default Button;