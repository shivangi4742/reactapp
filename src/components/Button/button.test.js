import React from 'react';
import { shallow } from 'enzyme';
import Button from './button';
import toJson from 'enzyme-to-json';
import DS from '../../utils/design-system';
const mockFn = jest.fn();

describe('Button', () => {
   
    it('is defined', () => {
        const button = shallow(<Button />);
        expect(button).toBeDefined();
    });
    it('matches the snapshots of Button component', () => {
        const tree = shallow(<Button />);
        expect(toJson(tree)).toMatchSnapshot();
    });
    it('render the primary button', () => {
        const button = shallow(<Button />);
        expect(button).toMatchSnapshot();
    });
    it('render the secondary button', () => {
        const button = shallow(<Button />);
        expect(button.html().match(/style="([^"]*)"/i)[1]).toBe(`color: ${DS.THEME.SECONDARY.main};`);
    });
    it('simulate click', () => {
        const button = shallow(<Button />);
        button.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });
})