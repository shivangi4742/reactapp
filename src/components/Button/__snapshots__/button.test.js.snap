// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches the snapshots of Button component 1`] = `
<styled.button
  className="customBtn"
>
   
</styled.button>
`;

exports[`Button render the primary button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        " ",
      ],
      "className": "customBtn",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      " ",
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    border: none;
    background: transparent;
    text-align: center;
    border-radius: ",
          "4px",
          ";
    border: 1px solid ",
          "
    color: ",
          ";
    width: ",
          "140",
          "px;
    height: ",
          "32",
          "px;
    line-height: ",
          "32",
          "px;
    font-size: ",
          "16",
          "px;
    text-transform: capitalize;
   ",
          [Function],
          "
   ",
          [Function],
          "
   ",
          [Function],
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          " ",
        ],
        "className": "customBtn",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " ",
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
    border: none;
    background: transparent;
    text-align: center;
    border-radius: ",
            "4px",
            ";
    border: 1px solid ",
            "
    color: ",
            ";
    width: ",
            "140",
            "px;
    height: ",
            "32",
            "px;
    line-height: ",
            "32",
            "px;
    font-size: ",
            "16",
            "px;
    text-transform: capitalize;
   ",
            [Function],
            "
   ",
            [Function],
            "
   ",
            [Function],
            "
    ",
            [Function],
            "
    ",
            [Function],
            "
    ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
